From f3a1d2f77d4cb018209cae55f7eef910d682d4d3 Mon Sep 17 00:00:00 2001
From: Pierre-Henry Moussay <pierre-henry.moussay@microchip.com>
Date: Fri, 28 Jun 2024 01:38:50 +0100
Subject: [PATCH] riscv: dts: microchip: add AMP support to pic64gx curiosity
 kit

Add node to support IHC v2 IP and remote proc to enable AMP capability

Signed-off-by: Pierre-Henry Moussay <pierre-henry.moussay@microchip.com>
---
 .../pic64gx-curiosity-kit-context-a.dts       | 181 ++++++++++++++++++
 1 file changed, 181 insertions(+)
 create mode 100644 arch/riscv/boot/dts/microchip/pic64gx-curiosity-kit-context-a.dts

diff --git a/arch/riscv/boot/dts/microchip/pic64gx-curiosity-kit-context-a.dts b/arch/riscv/boot/dts/microchip/pic64gx-curiosity-kit-context-a.dts
new file mode 100644
index 000000000000..ad91b347b2ca
--- /dev/null
+++ b/arch/riscv/boot/dts/microchip/pic64gx-curiosity-kit-context-a.dts
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/* Copyright (c) 2020-2021 Microchip Technology Inc */
+
+/dts-v1/;
+
+#include "pic64gx.dtsi"
+
+#include "dt-bindings/mailbox/miv-ihc.h"
+
+/* Clock frequency (in Hz) of the rtcclk */
+#define RTCCLK_FREQ            1000000
+
+/ {
+	#address-cells = <2>;
+	#size-cells = <2>;
+	model = "Microchip Pic64GX Curiosity Kit";
+	compatible = "microchip,pic64gx-curiosity-kit", "microchip,pic64gx";
+
+	aliases {
+		ethernet0 = &mac0;
+		serial1 = &mmuart1;
+	};
+
+	chosen {
+		stdout-path = "serial1:115200n8";
+	};
+
+	cpus {
+		timebase-frequency = <RTCCLK_FREQ>;
+	};
+
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x0 0x80000000 0x0 0x40000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		hss: hss-buffer@bfc00000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0xbfc00000 0x0 0x400000>;
+			no-map;
+		};
+
+		contextb_reserved: contextb_reserved@91c00000 {
+			reg = <0x0 0x91c00000 0x0 0x100000>;
+			no-map;
+		};
+
+		vdev0vring0: vdev0vring0@91d00000 {
+			reg = <0x0 0x91d00000 0x0 0x8000>;
+			no-map;
+		};
+
+		vdev0vring1: vdev0vring1@91d08000 {
+			reg = <0x0 0x91d08000 0x0 0x8000>;
+			no-map;
+		};
+
+		vdev0buffer: vdev0buff@91d10000 {
+			compatible = "shared-dma-pool";
+			reg = <0x0 0x91d10000 0x0 0x40000>;
+			no-map;
+		};
+
+		rsctable: rsc-table@91d50000 {
+			reg = <0x0 0x91d50000 0x0 0x1000>;
+			no-map;
+		};
+	};
+
+	periph-bus@40000000 {
+		compatible = "simple-bus";
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges = <0x0 0x40000000 0x0 0x40000000 0x0 0x20000000>,
+			 <0x0 0x60000000 0x0 0x60000000 0x0 0x20000000>,
+			 <0x0 0xe0000000 0x0 0xe0000000 0x0 0x20000000>,
+			 <0x20 0x0 0x20 0x0 0x10 0x0>,
+			 <0x30 0x0 0x30 0x0 0x10 0x0>;
+
+		mailbox@50000000 {
+			compatible = "microchip,miv-ihc-rtl-v2";
+			reg = <0x0 0x50000000 0x0 0x1C000>;
+			interrupt-parent = <&plic>;
+			interrupts = <180>, <179>, <178>;
+			interrupt-names = "hart-1", "hart-2", "hart-3";
+			#mbox-cells = <1>;
+			status = "disabled";
+		};
+	};
+
+	rproc_contextb: remote-context {
+		compatible = "microchip,ipc-remoteproc";
+		memory-region = <&vdev0buffer>, <&rsctable>,
+				<&contextb_reserved>, <&vdev0vring0>,
+				<&vdev0vring1>;
+		mboxes= <&ihc 8>;
+		status = "disabled";
+	};
+
+	ihc: mailbox {
+		compatible = "microchip,sbi-ipc";
+		interrupt-parent = <&plic>;
+		interrupts = <180>, <179>, <178>;
+		interrupt-names = "hart-1", "hart-2", "hart-3";
+		#mbox-cells = <1>;
+		status = "disabled";
+	};
+
+};
+
+&mac0 {
+	status = "okay";
+	phy-mode = "sgmii";
+	phy-handle = <&phy0>;
+
+	phy0: ethernet-phy@b {
+		reg = <0xb>;
+	};
+};
+
+&mbox {
+	status = "okay";
+};
+
+&mmc {
+	bus-width = <4>;
+	disable-wp;
+	cap-sd-highspeed;
+	cap-mmc-highspeed;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	no-1-8-v;
+	status = "okay";
+};
+
+&mmuart1 {
+	status = "okay";
+};
+
+&refclk {
+	clock-frequency = <125000000>;
+};
+
+&rtc {
+	status = "okay";
+};
+
+&syscontroller {
+	status = "okay";
+};
+
+&gpio0 {
+	status ="okay";
+	gpio-line-names =
+		"", "", "", "", "", "", "", "",
+		"", "", "", "", "MIPI_CAM_RESET", "MIPI_CAM_STANDBY";
+};
+
+&gpio1 {
+	status ="okay";
+	gpio-line-names =
+		"", "", "LED1", "LED2", "LED3", "LED4", "LED5", "LED6",
+		"LED7", "LED8", "", "", "", "", "", "",
+		"", "", "", "", "HDMI_HPD", "", "", "GPIO_1_23";
+};
+
+&gpio2 {
+	status ="okay";
+	gpio-line-names =
+		"", "", "", "", "", "", "SWITCH2", "USR_IO12",
+		"DIP1", "DIP2", "", "DIP3", "USR_IO1", "USR_IO2", "USR_IO7", "USR_IO8",
+		"USR_IO3", "USR_IO4", "USR_IO5", "USR_IO6", "", "", "USR_IO9", "USR_IO10",
+		"DIP4", "USR_IO11", "", "", "SWITCH1", "", "", "";
+};
\ No newline at end of file
-- 
2.30.2

